.eslintignore

```diff
@@ -13,3 +13,6 @@ doc/changelogs/CHANGELOG_v1*.md
 !doc/changelogs/CHANGELOG_v18.md
 !doc/api_assets/*.js
 !.eslintrc.js
+# These test --experimental-detect-module so contains ESM syntax in .js files.
+test/es-module/test-require-module-detect-entry-point.js
+test/es-module/test-require-module-detect-entry-point-aou.js
```

Comment:
You could use `spawnPromisified` like the other `detect-module` tests and then there’s no need to make lint exceptions.

Reply:
That would make the tests harder and slower to run. I prefer to spawn as few child processes as possible (also, spawnPromisified is a very inadequate utility at this point, it doesn't automatically log the stdout and stderr when the process fails, making debugging the CI logs a lot harder compared to the utilities we have in `common/child_process`).

Reply:
> That would make debugging the tests harder

I don’t find this to be a consensus opinion; there are several of us who have written tests using `spawnPromisified`, and I don’t find them hard to debug. A key is to use `assert.match` to check stdout/stderr, as it’ll print the expected and actual when they mismatch. You can follow the examples of the existing tests. And yes, using `assert.match` means that on failure, stdout or stderr are printed in the CI logs.

Reply:
> stdout or stderr are printed in the CI logs 

That is the problem: as someone who regularly checks flakes and debug them, I often need both stderr and stdout, not or.

> there are several of us who have written tests using spawnPromisified

How many of them work on CI flakes that are unrelated to their PRs?

Reply:
Which would lead to a lot of boilerplate code that are already implemented in common/child_process….

But this is already off-topic. The point is I find it harder to debug when any child process needs to be spawned at all.  Adding lint exceptions is worth the better debugging experience I get from direct logging of the test runner and clean stack traces  (even better, it outputs the entire command with all the necessary flags when the test fails for me to quickly copy-paste it into the debugger). Unless spawning a child process at the expense of this debugging experience is a blocking request, I will resolve this comment.

Reply:
You can add a less drastic lint exception: add a config override that changes the parse goal.

Reply:
Added a config override instead.